src/wallet/load.cpp

```diff
@@ -90,7 +90,11 @@ bool VerifyWallets(interfaces::Chain& chain)
 bool LoadWallets(interfaces::Chain& chain)
 {
     try {
+        std::set<fs::path> wallet_paths;
         for (const std::string& name : gArgs.GetArgs("-wallet")) {
+            if (!wallet_paths.insert(name).second) {
```

Comment:
```suggestion
            if (!wallet_paths.insert(fs::absolute(name, GetWalletDir()).second) {
```

you'll probably need something like this

Reply:
> you'll probably need something like this

This doesn't seem like this best thing. What is it supposed to do? The name specified `-wallet=<name>` is significant and shouldn't be normalized or transformed. It needs to be used as the RPC endpoint name, so for example must match the `bitcoin-cli -rpcwallet=<name>` string exactly and will be shown in `listwallets` and `getwalletinfo` exactly.

We should figure out what's causing the error you reported https://github.com/bitcoin/bitcoin/pull/20199#issuecomment-721714939. It seems promag and jonas tried to reproduce it but haven't been able, if I'm following correctly

Reply:
re: https://github.com/bitcoin/bitcoin/pull/20199#discussion_r517501836

> We should figure out what's causing the error you reported [#20199 (comment)](https://github.com/bitcoin/bitcoin/pull/20199#issuecomment-721714939). It seems promag and jonas tried to reproduce it but haven't been able, if I'm following correctly

I'd also be curious to know if that error is caused by this PR or happens without it. We know this PR is a reasonable tweak and narrow fix for a specific issue. So if there's some related problem that this PR doesn't fix, it could make sense to address separately.

Reply:
> What is it supposed to do?

It is simply using the same duplicate check when loading all wallet that is used previously when verifying all wallets. The name isn't normalized or touched at all. The absolute path is only used to check for duplicates.

> It seems promag and jonas tried to reproduce it but haven't been able

Jonas has reproduced it. I suspect the error message is different because jonas used a bdb wallet and I used a sqlite wallet.

> I'd also be curious to know if that error is caused by this PR or happens without it. 

The error won't happen without this pr because it will fail early if a duplicate wallet is specified.

Reply:
> > It seems promag and jonas tried to reproduce it but haven't been able
> 
> Jonas has reproduced it. I suspect the error message is different because jonas used a bdb wallet and I used a sqlite wallet.

Is this referring to https://github.com/bitcoin/bitcoin/pull/20199#issuecomment-721734126? The error Jonas reported there is a legitimate error that should *not* be a warning because the two wallet names are not the same and the wallets have different RPC endpoints. But it would be ideal to skip the warning in that case and only show the error. So that is a good point. I believe the solution would be to change `wallet_paths.insert(path)` to `wallet_paths.insert(wallet_file)` in the `VerifyWallets` function

Reply:
> Jonas has reproduced it. I suspect the error message is different because jonas used a bdb wallet and I used a sqlite wallet.

There are two different errors each related to the respective database type used.

When BDB is used (as Jonas did), the error is because of BDB's requirement of unique wallet files. It is opening the second wallet and seeing that it's BDB ID is one that is already loaded. So it gives the error that the wallet.dat file is already loaded.

When SQLite is used (as Marco did), the error is because an exclusive lock has been acquired on the database file already and another one cannot be acquired on the same file. It seems like that error message isn't being propagated up to the dialog, but clicking "Show details" will show the error.

Reply:
Just to add on, I think this PR does what it intended and is good in it's current form, but could be improved now or in the future to:

1. Check for duplicate `wallet_file` values instead of duplicate `path` values in `VerifyWallets()` to avoid pointlessly showing a warning dialog before showing a different error dialog as described https://github.com/bitcoin/bitcoin/pull/20199#issuecomment-721734126

2. Avoid the blank error dialog described https://github.com/bitcoin/bitcoin/pull/20199#issuecomment-721714939. Probably the GUI should default to showing the original string instead of an empty string if a translation isn't available. Also the "Unable to obtain an exclusive lock on the database, is it being used by another bitcoind?" text should probably be tweaked. It doesn't mention the filename, and in this case the lock is held by the same bitcoin-qt process not a different bitcoind process. Something like "Cannot open database file '{m_file_path}' which is already opened, perhaps in a different process"' might be better.

Reply:
Ok, clearly the wallet.dat files are the same, but I didn't know that they get different RPC endpoints (even when in the default `wallets` folder). Thanks for pointing that out.